pkg/tfgen/generate_schema.go

```diff
@@ -135,12 +131,16 @@ func (nt *schemaNestedTypes) declareType(path *paths.TypePath,
 		requiredOutputs = required
 	}
 
+	// Merging makes sure that structurally identical types are shared and not generated more than once.
```

Comment:
This is the interesting bit in the tfgen algorithm! Adding a comment to highlight it.